### ISSUE TITLE
How to revert back to greenlight-v2?

### ISSUE THREAD

**Issue by dashohoxha** [bigbluebutton/bbb-install#646]
Maybe `greenlight-v3` is better than `greenlight-v2` for an initial installation (for reasons that I don't understand). However, as long as there is no clean and easy migration path from v2 to v3, I'd like to stick to `greenlight-v2`.

I still have the old directory `greenlight/`, which I assume preserves all the necessary data. So, I believe that it should be possible to switch back to `greenlight-v2`. Any advice on how to do it? I don't think that it should be too difficult.

**Comment by ffdixon** [comment]
>  which I assume preserves all the necessary data. So, I believe that it should be possible to switch back to greenlight-v2

Right.  You should be able to start/stop either greenlight with their associated `docker-compose` command.

**Comment by dashohoxha** [comment]
> You should be able to start/stop either greenlight with their associated `docker-compose` command.

I tried these:

``` bash
cd /root/greenlight-v3/
docker-compose down
cd /root/greenlight/
docker-compose up -d
```

Then I had to fix the nginx configuration:
1. On `/etc/nginx/sites-available/bigbluebutton` I replaced:

    ``` nginx
      # BigBlueButton landing page.
      location / {
        root   /var/www/bigbluebutton-default/assets;
        try_files $uri @bbb-fe;
      }
    ```

    with:
    ``` nginx
      # BigBlueButton landing page.
      location = / {
        return 301 https://$server_name/b;
      }
      location /b {
        root   /var/www/bigbluebutton-default/assets;
        try_files $uri @bbb-fe;
      }
    ```

2. On the directory `/usr/share/bigbluebutton/nginx` I renamed `greenlight-v3.nginx` to `greenlight-v3.nginx.disabled`. Then modified `greenlight.nginx` by replacing `location /b {` with `location @bbb-fe {`.

3. Reload nginx: `systemctl reload nginx`.

For some reasons, I had also to update `BIGBLUEBUTTON_SECRET` on `/root/greenlight/.env`, and the rebuild it:

``` bash
cd /root/greenlight/
bbb-conf --secret
vim .env
docker-compose down
docker-compose up -d
```

It seems to work. I hope that I have not missed any other steps that are required.

Anyway, I guess that next time that I try to update (by running `bbb-install-2.6.sh`) I may have to repeat some of these steps again.

**Comment by dashohoxha** [comment]
> Then I had to fix the nginx configuration:

You know what? It is possible to run both gl-v2 and gl-v3 at the same time. The only nginx modification that is needed is this:

Edit the file `/usr/share/bigbluebutton/nginx/greenlight.nginx` and replace

``` nginx
location /b {
  proxy_pass          http://127.0.0.1:5000;
[ . . . . . . . . . . . . . . . ]

```

with this:

``` nginx
location /b {
  root   /var/www/bigbluebutton-default/assets;
  try_files $uri @bbb-gl-v2;
}

location @bbb-gl-v2 {
  proxy_pass          http://127.0.0.1:5000;
[ . . . . . . . . . . . . . . . ]

```

Of course, both of GL apps need to be started:

``` bash
cd /root/greenlight/
docker-compose up -d

cd /root/greenlight-v3/
docker-compose up -d
```

After the migration from v2 to v3 is done properly, I can shutdown gl-v2.

@ffdixon  I would suggest that the installation script (`bbb-install-2.6.sh`) enables both of the GL versions by default.

**Comment by dashohoxha** [comment]
> Edit the file /usr/share/bigbluebutton/nginx/greenlight.nginx and replace

Actually this modification is not needed either. So, no modifications at all on nginx. Just update the `BIGBLUEBUTTON_SECRET` on `/root/greenlight/.env` and then start up the gl-v2 app. Then both versions of the greenlight will be available at the same time.

**Comment by paulmenzel** [comment]
Thank you for documenting all your steps. As running greenlight-v2 seems to work, is there still something to be improved in the install script in this regard?

**Comment by lmsofindia** [comment]
hi @dashohoxha - could you please help me here...
I have installed BBB 2.7 and then trying use greenlight v2...
While installing BBB, i installed GL V3 which I do not want to use..

I tried your steps above but when https://abced.bbb.in/b - > it gives me page is not redirecting properly

Find below my changes
#1 - in /etc/nginx/sites-available/bigbluebutton
![image](https://github.com/user-attachments/assets/85f92a1f-8a60-4cd7-8415-8e69e5feab64)

#2 - in /usr/share/bigbluebutton/ngix
![image](https://github.com/user-attachments/assets/33728c53-8891-4fbb-8115-fd84eab61c2d)

inside greenlight.ngix

![image](https://github.com/user-attachments/assets/1323a4e6-b9eb-4984-935a-eb1e1c8499de)

#3 - i have kept the same secret in .env  also

Please help

**Comment by ffdixon** [comment]
@lmsofindia We've not done any development on GreenLight V2 for over a year now.  Is it an option for you to use V3?

**Comment by dashohoxha** [comment]
@lmsofindia After looking at my comments (because I don't remember anymore), it seems that my instructions are these:

> Just update the BIGBLUEBUTTON_SECRET on /root/greenlight/.env and then start up the gl-v2 app. Then both versions of the greenlight will be available at the same time.

However this worked for the version 2.6, I have not tried it with the version 2.7, because I managed somehow to migrate everything to gl-v3 and then removed gl-v2.

So, basically, don't make any changes to configuration files. Just get the secret with `bbb-conf --secret`  and update `/root/greenlight/.env`. Then start up gl-v2 (`docker compose up -d`). If you are lucky it should work.

**Comment by dashohoxha** [comment]
> Is it an option for you to use V3?

@ffdixon In my case the problem was that there was no clean migration path from V2 to V3, so using V3 was not an option, although it might be better or more beautiful.

I don't know what is the problem for @lmsofindia

**Comment by dashohoxha** [comment]
> #3 - i have kept the same secret in .env also

@lmsofindia You have to update the secret with the new one, I guess, and then rebuild the container.

**Comment by ffdixon** [comment]
Does this enable you to migrate?

  https://docs.bigbluebutton.org/greenlight/v3/migration/

**Comment by lmsofindia** [comment]
hi @ffdixon - thank you so much for all your help!. I actually tried V3.. but the keycloak is I think little messed up.. just to share the feedback

a) As per the steps, i could set up K fine first time...and there was an admin of Keycloak account created and google auth also set up fine.
But , as ours is migration (v2 to v3 all migration steps ran fine ), even after following steps of KEY_BASE and all those, none of our users were able to login in greenlight v3... ..(before keycloak, local users were able to login even in v3)... after after K they could not..

b) Then we had to restart our bbb for some reason, 

c) Then /keycloak - admin account can't login at all...
it always gives the error " admin user already exists , can not be created",.... i tried all help available in various forums by changing docker-compose, /down.. up.... changing new admin account name...

nothing actually works..

More over we use shibboleth saml/ldap so at last we decided to go back to v2 again..

====

UPDATE:
I am able to make greenlight v2 up and running...

the solution is - proxy_set_header X-Forwarded-Ssl on; in the greenlight.nginx

====
Generic feedback - I think external authentication v2 is simple and better.... simple solutions are better for large scale implementation and managing also takes less time..

We use BBB and Greenlight for a user base of more than 5000 ... for an university via moodle as well as direct greenlight..
Thanks for the great software.

Appreciate all your help!

Thanks!
Mihir

**Comment by harika1258** [comment]
challenge after a week installing bbb 2.7 and gl 2 and disappointing, I found your post.  
Thank you so much @dashohoxha  and @lmsofindia
